src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml

```diff
@@ -2441,13 +2441,21 @@ module Ledger = struct
       (fun () -> failwith "invalid scalar")
       Fn.id
 
+  let token_id_checked (token : field_class Js.t) =
+    token |> of_js_field |> Mina_base.Token_id.Checked.of_field
+
   let token_id (token : field_class Js.t) : Mina_base.Token_id.t =
-    token |> of_js_field |> to_unchecked |> Mina_base.Token_id.of_field
+    token |> of_js_field_unchecked |> Mina_base.Token_id.of_field
 
   let default_token_id_js =
     Mina_base.Token_id.default |> Mina_base.Token_id.to_field_unsafe
     |> Field.constant |> to_js_field
 
+  let account_id_checked pk token =
+    Mina_base.Account_id.Checked.create
+      (Signature_lib.Public_key.Compressed.var_of_t (public_key pk))
```

Comment:
same applies here about `public_key` + `var_of_t` not being the approach we want